packages/library-base/src/commonMain/kotlin/io/realm/kotlin/internal/util/CoroutineDispatcherFactory.kt

```diff
@@ -0,0 +1,63 @@
+package io.realm.kotlin.internal.util
+
+import io.realm.kotlin.internal.platform.multiThreadDispatcher
+import io.realm.kotlin.internal.platform.singleThreadDispatcher
+import kotlinx.coroutines.CloseableCoroutineDispatcher
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+
+/**
+ * Factory wrapper for passing around dispatchers without needing to create them. This makes it
+ * possible to configure dispatchers in Configuration objects, but allow Realm instances to control
+ * their lifecycle.
+ */
+public class CoroutineDispatcherFactory private constructor(
+    private val isManaged: Boolean,
+    private val createDispatcher: () -> CoroutineDispatcher,
+) {
+    public companion object {
+        @OptIn(ExperimentalCoroutinesApi::class)
+        public fun internal(name: String, threads: Int = 1): CoroutineDispatcherFactory {
```

Comment:
Not really. I just wanted to future-proof the API since we have constructs in our code for creating multithreaded dispatchers. But I can remove it again if it is confusing.